### PR TITLE
Ludown: Batching calls to translate service. Improves ludown translate perf by 10x

### LINKED ISSUE #695: ludown tool should batch calls to translator
## Tool
Name: Ludown
OS: Windows

## Describe the bug
When using ludown translate each utterance requires a roundtrip to the translator resulting in long translation times for large files and failures in some instances blocking use.

## Expected behavior

https://docs.microsoft.com/en-us/azure/cognitive-services/translator/reference/v3-0-translate?tabs=curl outlines a batch translation option. A batched array of utterances could be passed to the translator in one call. ludown would then match originalText[i] to translatedText[i] and write to the file.

A CLI parameter would enable the user to control the batch size.

## Screenshots

image

[bug]

### PR DESCRIPTION
**Pull request by vishwacsena**
Fixes #695 and #642 

## Proposed Changes
- Implemented batching for calls to translation service
- Provided an over-ride for customers to set a lower batch size (if required)
- Improves translate perf by up to 10x

## Testing
- Updated existing tests
- No new tests added since there is no functional change.

### CODE REVIEW

**Comment by cleemullins** [comment]
@vishwacsena The Travis build is failing for this PR. Are you able to track that down?

File: packages/Ludown/docs/translate-command.md [hunk 1]
```diff
@@ -37,6 +37,7 @@ When translating .lu file,
     -n, --lu_File <LU_File>          [Optional] Output .lu file name
     -c, --transate_comments          [Optional] Translate comments in .lu files
     -u, --translate_link_text        [Optional] Translate URL or .lu file reference link text
+    -x, --batch_translate <linesToBatch>  [Optional] Batch up <x> (1-25) number of lines before calling translation API. Defaults to 25.
```
Comment (cleemullins) [inline packages/Ludown/docs/translate-command.md]: How about "-b" for match. using "-x" seems weird.

File: packages/Ludown/docs/commands.md [hunk 2]
```diff
@@ -141,6 +141,7 @@ To bootstrap translations of the language understanding content for your bot, yo
     -n, --lu_File <LU_File>          [Optional] Output .lu file name
     -c, --transate_comments          [Optional] Translate comments in .lu files
     -u, --translate_link_text        [Optional] Translate URL or .lu file reference link text
+    -x, --batch_translate <linesToBatch>  [Optional] Batch up <x> (1-25) number of lines before calling translation API. Defaults to 25.
```
Comment (cleemullins) [inline packages/Ludown/docs/commands.md]: How about "-b" for match. using "-x" seems weird.

File: packages/Ludown/lib/ludown-translate.js [hunk 3]
```diff
@@ -26,9 +26,16 @@ program
     .option('-n, --lu_File <LU_File>', '[Optional] Output .lu file name')
     .option('-c, --translate_comments', '[Optional] Translate comments in .lu files')
     .option('-u, --translate_link_text', '[Optional] Translate URL or .lu file reference link text')
+    .option('-x, --batch_translate <linesToBatch>', '[Optional] Batch up <x> (1-25) number of lines before calling translation API. Defaults to 25.')
```
Comment (cleemullins) [inline packages/Ludown/lib/ludown-translate.js]: How about "-b" for match. using "-x" seems weird.

File: packages/Ludown/docs/commands.md (hunk 2 above)
Comment (vishwacsena) [reply packages/Ludown/docs/commands.md]: Yes. switched to -b.

**Pull request merged**